### PR TITLE
[MM-42742] Top Reactions Documentation

### PR DESCRIPTION
**Pull request by mylonsuren**
#### Summary

Adds documentation for the top reactions for team/user endpoint. This endpoint is currently hidden behind a feature flag.

**See server PR for additional context:**
- https://github.com/mattermost/mattermost-server/pull/19850

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-42742

### CODE REVIEW

**Review by mkraft — CHANGES_REQUESTED** [review]

File: v4/source/reactions.yaml [hunk 1]
```diff
@@ -27,6 +27,128 @@
...
           $ref: "#/components/responses/Forbidden"
+  "/teams/{team_id}/top/reactions":
+    get:
+      tags:
+        - reactions
+      summary: Get a list of the top reactions for a team
+      description: |
+        Get a list of the top reactions across all public channels, and private channels (that the user is a member of) for a given team.
+        ##### Permissions
+        Must have insights enabled.
```
[the remark this answers is not in the document]
Comment (mylonsuren) [reply v4/source/reactions.yaml]: Ok that makes sense, I had included it just to be safe, but I've removed it now so should be good.

**Review by mkraft — APPROVED** [review]

File: v4/source/definitions.yaml [hunk 2]
```diff
@@ -771,6 +771,27 @@ components:
           description: The time in milliseconds this reaction was made
           type: integer
           format: int64
+    TopReaction:
+      type: object
+      properties:
+        emoji_name:
+          description: The name of the emoji that was used for this reaction
```
Comment (cwarnermm) [inline v4/source/definitions.yaml:778]:
```suggestion
          description: The name of the emoji used for this reaction.
```

File: v4/source/reactions.yaml [hunk 3]
```diff
@@ -27,6 +27,124 @@
...
+          schema:
+            type: string
+        - name: time_range
+          in: query
+          description: >
+            Time Range can be "today", "7_day" or "28_day".
+
+            - `today`: reactions posted on the current day
+            
+            - `7_day`: reactions posted in the last 7 days
```
Comment (cwarnermm) [inline v4/source/reactions.yaml:108]:
```suggestion
            - `7_day`: reactions posted in the last 7 days.
```

File: v4/source/reactions.yaml [hunk 4]
```diff
@@ -27,6 +27,124 @@
...
+            
+            - `28_day`: reactions posted in the last 28 days
```
Comment (cwarnermm) [inline v4/source/reactions.yaml:110]:
```suggestion
            - `28_day`: reactions posted in the last 28 days.
```

File: v4/source/reactions.yaml [hunk 5]
```diff
@@ -27,6 +27,124 @@
...
+            type: string
+        - name: page
+          in: query
+          description: The page to select.
+          schema:
+            type: integer
+            default: 0
+        - name: per_page
+          in: query
+          description: The number of reactions per page. There is a maximum limit of 200 reactions per page.
```
Comment (cwarnermm) [inline v4/source/reactions.yaml:122]:
```suggestion
          description: The number of reactions per page, up to a maximum of 200.
```

File: v4/source/reactions.yaml [hunk 6]
```diff
@@ -27,6 +27,124 @@
...
+            Team ID will scope the response to a given team.
+
+            ##### Permissions
+
+            Must have `view_team` permission for the team.
+          schema:
+            type: string
+      responses:
+        "200":
+          description: Top reactions retrieve successful
```
Comment (cwarnermm) [inline v4/source/reactions.yaml:138]: Could rewording this to "Top reactions retrieved successfully." work?

File: v4/source/reactions.yaml [hunk 7]
```diff
@@ -27,6 +27,124 @@
...
+                $ref: "#/components/schemas/TopReactionList"
+        "400":
+          $ref: "#/components/responses/BadRequest"
+        "403":
+          $ref: "#/components/responses/Forbidden"
+  "/users/me/top/reactions":
+    get:
+      tags:
+        - reactions
+      summary: Get a list of the top reactions for a user
```
[the remark this answers is not in the document]
Comment (cwarnermm) [reply v4/source/reactions.yaml:88]: Please review for technical accuracy. I attempted to write this in a parallel way to the teams equivalent.

File: v4/source/reactions.yaml [hunk 8]
```diff
@@ -27,6 +27,124 @@
...
+        - name: user_id
+          in: path
+          description: User GUID
+          required: true
+          schema:
+            type: string
+        - name: time_range
+          in: query
+          description: >
+            Time Range can be "today", "7_day" or "28_day".
```
Comment (cwarnermm) [inline v4/source/reactions.yaml:104]:
```suggestion
            Time range can be "today", "7_day", or "28_day".
```

**Review by cwarnermm — COMMENTED** [review]
Thank you, @mylonsuren!, for this great docs update! I've provided feedback inline. The majority of feedback is on punctuation consistency.

File: v4/source/definitions.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (mylonsuren) [reply v4/source/definitions.yaml]: Yes, they're both referring to the same field.

File: v4/source/reactions.yaml (hunk 7 above)
[the remark this answers is not in the document]
Comment (mylonsuren) [reply v4/source/reactions.yaml:88]: Yes that makes sense, thanks for adding the extra detail.

**Comment by mylonsuren** [comment]
@cwarnermm The changes you've requested have been added now, thanks for the feedback! Just double checking that this is OK to merge?

**Comment by cwarnermm** [comment]
Thanks @mylonsuren! We're ready to merge from my perspective.

**Pull request merged**